lib/ae_mdw_web/controllers/aexn_token_controller.ex

```diff
@@ -90,21 +90,22 @@ defmodule AeMdwWeb.AexnTokenController do
       query: query
     } = assigns
 
-    with {:ok, paginated_balances} <-
-           Aex9.fetch_event_balances(state, contract_id, pagination, cursor, order_by, query) do
+    with {:ok, contract_pk} <- validate_aex9(contract_id, state),
+         {:ok, paginated_balances} <-
+           Aex9.fetch_event_balances(state, contract_pk, pagination, cursor, order_by, query) do
```

Comment:
Why are we moving validations back to the controller? Let's leave them on the context modules, validations, rendering and paginating.